app/controllers/api/v1/merchants_controller.rb

```diff
@@ -4,7 +4,13 @@ def index
   end
 
   def show
-    render json: MerchantSerializer.new(Merchant.find(params[:id]))
+    if params[:item_id]
+      item = Item.find(params[:item_id])
+      merchant = item.merchant
+    else
+      merchant = Merchant.find(params[:id])
+    end
```

Comment:
this handles whether to show the merchant for a given item, or to just show the merchant that had its id passed it as a param.